pkg/controllers/controller.go

```diff
@@ -677,6 +679,8 @@ func (h *handler) OnRegistrationChange(_ string, registrationObj *v1.Registratio
 		v1.ResourceConditionReady.False(updated)
 		v1.ResourceConditionDone.False(updated)
 		v1.RegistrationConditionActivated.False(updated)
+		// Set ResourceConditionProgressing as the CurrentCondition since we're resetting the registration process
+		updated.SetCurrentCondition(v1.ResourceConditionProgressing)
```

Comment:
Do we need to update the current condition on line 729?

Reply:
The `PrepareSuccessfulActivation` shared processor called on line 728 should cover setting the current status for us. 👍